doc/develop/sca/codechecker.rst

```diff
@@ -0,0 +1,58 @@
+.. _codechecker:
+
+CodeChecker support
+###################
+
+`CodeChecker <https://codechecker.readthedocs.io/>`__ is a static analysis infrastructure.
+It executes analysis tools available on the build system, such as
+`Clang-Tidy <http://clang.llvm.org/extra/clang-tidy/>`__,
+`Clang Static Analyzer <http://clang-analyzer.llvm.org/>`__ and
+`Cppcheck <https://cppcheck.sourceforge.io/>`__. Refer to the analyzer's websites for installation
+instructions.
+
+Installing CodeChecker
+**********************
+
+CodeChecker itself is a python package available on `pypi <https://pypi.org/project/codechecker/>`__.
+
+.. code-block:: shell
+
+    pip install codechecker
+
+Running with CodeChecker
+************************
+
+To run CodeChecker, :ref:`west build <west-building>` should be
+called with a ``-DZEPHYR_SCA_VARIANT=codechecker`` parameter, e.g.
+
+.. code-block:: shell
+
+    west build -b mimxrt1064_evk -s samples/basic/blinky -- -DZEPHYR_SCA_VARIANT=codechecker
+
+
+Configuring CodeChecker
+***********************
+
+To configure CodeChecker or analyzers used, arguments can be passed using the
+``CODECHECKER_ANALYZE_OPTS`` parameter, e.g.
+
+.. code-block:: shell
+
+    west build -b mimxrt1064_evk -s samples/basic/blinky -- -DZEPHYR_SCA_VARIANT=codechecker \
+    -DCODECHECKER_ANALYZE_OPTS="--config;$CODECHECKER_CONFIG_FILE;--timeout;60"
```

Comment:
I like the approach of doing it more generic with the opts.

I was just thinking if we can define an common SCA interface instead of having -D\<checker_name\>_\<xyz\> for each of the different checkers.

Suggestion:

- ZEPHYR_SCA_VARIANT:
  - as already defined by the current impl.
- ZEPHYR_SCA_ARGS:
  - any command line parameter (list) passed to the SCA executable
  - calls with sub commands can be distinguished by "@analyze;... ;... ;... ;@parse;... ;... ;" where the SCA integration impl. has to split the different args for each stage (sub-command)
  - e.g. -DZEPHYR_SCA_ARGS="@analyze;--config;$CODECHECKER_CONFIG_FILE;--timeout;60;@parse;--trim-path-prefix;$PWD"
  - That would only be required, if the same parameter can occur for same stages e.g. "@analyze;--timeout;60;@parse;--timeout;120", as long as the parameters are unambiguous concerning stages / sub-commands, no extra @\<stage name\> will be required. The SCA integration impl. can take care to assign each command line param to the right stage. 
- ZEPHYR_SCA_CFGFILE:
  - path to one or more config files (again list, if requried)
  - if more than one is required, the order of config files can define which file is for which stage or the same @\<stage name\> mechanism could be used.

I would still provide the ZEPHYR_SCA_CFGFILE as convenience variant to ZEPHYR_SCA_ARGS.
- Writing long command line parameter lists is cumbersome (I would avoid using it)
- Configuration files should be able to hold all parameter configurable via command line
- Configuration files can be stored in the source tree and are therefore provided and accessible to anyone using the checker.

Reply:
Not sure the common SCA arguments would be such a benefit, different tools won't be compatible.
The `ZEPHYR_SCA_ARGS` would need to be parsed, expanded, forwarded, ... This would add complexity and reduces maintainability. I like [KISS](https://en.wikipedia.org/wiki/KISS_principle).

Reply:
I like KISS too, but from a users perspective. Ending up with all kind of <MY-CHECKER>_<A-STAGE>_OPTS I considered not KISS for the user. The @<stage> was just a work-around to when I need to pass equally named opts with different values to different stages, which I think would be rare. Otherwise you just get one ZEPHYR_SCA_OPTS with all opts for all stages where the SCA integration needs to handle the opts assignment to the stages. Less KISS for the implementer and if command line options change, will needs maintenance - a clear down side.

In a discussion with @simhein he came up with the idea to integrate a SCA based on and reusing your codechecker integration by passing a configuration which checks the Zephyr project guideline rule.
In the end we would get something like e.g. ZEPHYR_SCA_VARIANT=codechecker-zrules without the need for the user to pass any opts.

Hence I withdraw my suggestion.